### PR TITLE
Update links to developer docs

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review. Can be approved in draft form.

## Description of Changes

Part of https://github.com/freedomofpress/securedrop/issues/6520. See https://github.com/freedomofpress/securedrop/issues/6520#issuecomment-1253044091 for context on the switch-over.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +14,13 @@ SecureDrop is an open-source whistleblower submission system that media organiza
 
 ## Documentation
 
-SecureDrop's documentation is built and hosted by [Read the Docs](https://readthedocs.org) at https://docs.securedrop.org. It is maintained in a standalone repository: https://github.com/freedomofpress/securedrop-docs
+SecureDrop's end user documentation is hosted at https://docs.securedrop.org. It is maintained in a standalone repository: https://github.com/freedomofpress/securedrop-docs
```
Comment (legoktm) [inline README.md]: Preexisting issue, but this sentence doesn't end with a period.

File: README.md [hunk 2]
```diff
@@ -14,10 +14,13 @@ SecureDrop is an open-source whistleblower submission system that media organiza
...
 
 By default, the documentation describes the most recent SecureDrop release. This is known as the **stable** version and is recommended for end users (Sources, Journalists, or Administrators). The **latest** documentation is automatically built from the most recent commit to the SecureDrop documentation repository. It is most useful for developers and contributors to the project. You can switch between versions of the documentation by using the toolbar in the bottom left corner of the Read the Docs screen.
 
+Developer documentation can be found at https://developers.securedrop.org/, maintained in
+https://github.com/freedomofpress/securedrop-dev-docs/
```
Comment (legoktm) [inline README.md]: Same here, no period.

**Review by legoktm — COMMENTED** [review]
Very minor nitpick.

File: README.md (hunk 1 above)
Comment (eloquence) [reply README.md]: Part of me never trusts that parsers will exclude periods from auto-links so I tend to superstitiously avoid them after a URL :). Fixed.

File: README.md (hunk 2 above)
Comment (eloquence) [reply README.md]: Fixed.

**Review by legoktm — DISMISSED** [review]

**Review by legoktm — APPROVED** [review]

**Pull request merged**